crates/bevy_ecs/src/world/entity_access/world_mut.rs

```diff
@@ -1658,7 +1706,7 @@ impl<'w> EntityWorldMut<'w> {
     /// This is *only* required when using the unsafe function [`EntityWorldMut::world_mut`],
     /// which enables the location to change.
     pub fn update_location(&mut self) {
-        self.location = self.world.entities().get(self.entity);
+        self.location = self.world.entities().get(self.entity).expect("Commands should not be queued which despawn an entity while an active `EntityWorldMut` is available to do so.");
```

Comment:
> Commands should not be queued which despawn an entity while an active `EntityWorldMut` is available to do so.

I think this should be a bit more functionally descriptive. Ex: "Attempted to update the location of a despawned entity, which is impossible. This was the result of performing an operation on this EntityWorldMut that queued a despawn command".